lib/Pool.js

```diff
@@ -339,6 +372,7 @@ class Pool {
     return new Promise((resolve, reject) => {
       odbc.connect(connectionConfig, (error, nativeConnection) => {
         if (error) {
+          this.pool.connectionEmitter.emit('connectionError', error);
           reject(error);
```

Comment:
So now we're emitting an error which is then handled by the new code above and returned if a user is asking for a connection and also rejecting the current promise, which gets handled somewhere else?

Could this lead to duplication of errors at all?

Reply:
It gets swallowed on lines 51-54:

https://github.com/markdirish/node-odbc/blob/a7b2b8e44d8c75df4a17f4bccc7dc9c65461a07f/lib/Pool.js#L48-L54

So while it rejects, it is simply to close the promise we created, not to actually throw/handle the error. My fear about doing it any other way is errors occurring when a connection is made (for whatever reason), but since the user doesn't have any user functions waiting for that promise, it just throws an uncatchable error and crashes the process. Instead, this will swallow the error and only produce catchable errors, which happen:

1. When a user creates the pool and there was a problem connecting the first connection (no network, bad connection string, etc.).
2. The user is trying to do some work and there isn't an open connection, so a connection is made, fails, and returns the error to the waiting user.

I wish there would be a friendly way to notify the developer of connection failures without crashing the application, but by the nature of connections taking place in the background, I'm not sure if there is a way. I don't think this is the _best_ solution, but the code is getting pretty convoluted as is, all in the name of performance when spinning up the pool.

Reply:
Ok, that makes sense. Might be worth some comments since it's quite complex.